types/dd-trace/index.d.ts

```diff
@@ -147,3 +151,34 @@ interface TraceOptions {
      */
     tags?: { [key: string]: any } | string;
 }
+
+declare class ScopeManager {
+    /**
+     * Get the current active scope or null if there is none.
+     *
+     * @todo The dd-trace source returns null, but opentracing's childOf span
+     *       option is typed as taking undefined or a scope, so using undefined
+     *       here instead.
+     */
+    active(): Scope | undefined;
```

Comment:
Should be `null` not `undefined`

Reply:
You’re right that dd-trace returns the literal `null`, it’s just that the opentracing API doesn’t actually accept `null` https://github.com/opentracing/opentracing-javascript/blob/0f145548a8c26f7761d33d274e5988e69ce2251e/src/tracer.ts#L14.

Having said that, I agree it seems cleaner to return literal `null` than `undefined`, so I’ll send a PR to opentracing instead, but it will probably take a while before it’s released and datadog uses it, so until then you’ll have to do `{ childOf: span ? span : undefined }`, which is a bit ugly.

Alternatively we can leave this typing as `undefined` for now and once the opentracing PR has been accepted, released, and in use in dd-trace, we can update this to specify `null` instead. What do you think?

Reply:
Hmm, the opentracing source goes out of its way to delete the `childOf` key from the options object. On reflection, that’s probably because `undefined` values get left out when converted to JSON, so it seems like really dd-trace should return `undefined` instead.

```
> JSON.stringify({ foo: undefined, bar: null })
< "{\"bar\":null}"
```

Reply:
Ah, okay. I'm fine leaving this as is.

Reply:
(I’ll make that PR to dd-trace as part of some PRs I’m making anyways)